testsuite/MDAnalysisTests/coordinates/test_memory.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+from numpy.testing import raises
+
+import MDAnalysis as mda
+from MDAnalysisTests.datafiles import DCD, PSF
+from MDAnalysisTests.coordinates.base import (BaseReference,
+                                              BaseReaderTest)
+from MDAnalysis.coordinates.memory import MemoryReader
+from numpy.testing import assert_equal
+
+
+class MemoryReference(BaseReference):
+    def __init__(self):
+        super(MemoryReference, self).__init__()
+        self.topology = PSF
+        self.trajectory = DCD
+        self.universe = mda.Universe(PSF, DCD)
+
+        self.n_atoms = self.universe.trajectory.n_atoms
+        self.n_frames = self.universe.trajectory.n_frames
+
+        self.dt = self.universe.trajectory.ts.dt
+        self.dimensions = self.universe.trajectory.ts.dimensions
+        self.totaltime = self.universe.trajectory.totaltime
+        self.volume = self.universe.trajectory.ts.volume
+
+        self.first_frame = MemoryReader.MemoryTimestep(self.n_atoms)
+        self.first_frame.positions = np.array(self.universe.trajectory[0])
+        self.first_frame.frame = 0
+        self.first_frame.time = self.first_frame.frame*self.dt
+
+        self.second_frame = MemoryReader.MemoryTimestep(self.n_atoms)
+        self.second_frame.positions = np.array(self.universe.trajectory[1])
+        self.second_frame.frame = 1
+        self.second_frame.time = self.second_frame.frame*self.dt
+
+        self.last_frame = MemoryReader.MemoryTimestep(self.n_atoms)
+        self.last_frame.positions = \
+            np.array(self.universe.trajectory[self.n_frames - 1])
+        self.last_frame.frame = self.n_frames - 1
+        self.last_frame.time = self.last_frame.frame*self.dt
+
+        self.jump_to_frame = self.first_frame.copy()
+        self.jump_to_frame.positions = np.array(self.universe.trajectory[3])
+        self.jump_to_frame.frame = 3
+        self.jump_to_frame.time = self.jump_to_frame.frame*self.dt
+
+
+    def reader(self, trajectory):
+        return mda.Universe(self.topology,
+                            trajectory, in_memory=True).trajectory
+
+    def iter_ts(self, i):
+        ts = self.universe.trajectory[i]
+        return ts
+
+
+
+class TestMemoryReader(BaseReaderTest):
+    def __init__(self):
+        reference = MemoryReference()
+        self.ref = reference
+        self.reader = self.ref.reader(self.ref.trajectory)
```

Comment:
`BaseReaderTest` has a long init that cannot be skipped. Also, `BaseReaderTest.reader` cannot be overwritten. A working init would here be:

```python
def __init__(self):
    reference = MemoryReference()
    super(TestMemoryReader, self).__init__(reference)
```